src/node/state.ml

```diff
@@ -84,6 +84,11 @@ let apply_block state block =
       (fun results (hash, receipt) -> BLAKE2B.Map.add hash receipt results)
       state.recent_operation_receipts receipts in
   let applied_blocks = block :: state.applied_blocks in
+  (* TODO: we might want to use a monotonic clock so we don't get weird things like
+     negative block rates. *)
+  let timestamp = Unix.time () in
```

Comment:
Is second precise enough?

Reply:
We care about TPS (transactions per second), so this will probably be fine.